core/blockchain.go

```diff
@@ -2394,6 +2394,16 @@ func (bc *BlockChain) maintainTxIndex() {
 	}
 	defer sub.Unsubscribe()
 
+	// Launch the initial processing if chain is not empty. This step is
+	// useful in these scenarios that chain has no progress and indexer
+	// is never triggered.
+	head := rawdb.ReadHeadBlock(bc.db)
+	tail := rawdb.ReadTxIndexTail(bc.db)
+	if head != nil && tail != nil {
```

Comment:
remove `tail` check here

Reply:
If we removed the check here, we need to adjust the txindex test cases in [core/blockchain_test.go](core/blockchain_test.go), because `InsertReceiptChain` will also modify the `TxIndexTail`

https://github.com/ethereum/go-ethereum/blob/4273f248755a48d29002aa6d01de94bca36ea6c1/core/blockchain.go#L1251-L1260


IMO, the purpose of introducing this PR is to continue the indexing/unindexing after the node restarts(even if no new chain is updated)

If `tail` is null, then it means the node is just inited and all blocks are not indexed yet or no blocks were inserted. In this case, We need to wait for at least one new block to be added before we start the indexing/unindexing process.

Let me think about how to modify the test cases if needed.

Reply:
I added this comment because @rjl493456442 told me during review, that he thinks it's not necessary to have the `tail` check. You can also leave it in if that's simpler.

Reply:
I also didn't think of it before, but the unit test failed and found it possible to pass `tail != nil` check condition, so there is no need to change the unit test.